lib/Backend/GlobOpt.cpp

```diff
@@ -19069,6 +19107,30 @@ GlobOpt::OptHoistInvariant(
     IR::BailOutKind bailoutKind)
 {
     BasicBlock *landingPad = loop->landingPad;
+
+    IR::Opnd* src1 = instr->GetSrc1();
+    if (src1)
+    {
+        // We are hoisting this instruction possibly past other uses, which might invalidate the last use info. Clear it.
+        OptHoistToLandingPadUpdateValueType(landingPad, instr, src1, src1Val);
```

Comment:
It's the lowering that looked at the valueType and incorrectly omitted tag checks....

---
In reply to: [101592247](https://github.com/Microsoft/ChakraCore/pull/2538#discussion_r101592247) [](ancestors = 101592247,101557970)